c/driver/postgresql/connection.cc

```diff
@@ -654,10 +652,23 @@ AdbcStatusCode PostgresConnectionGetInfoImpl(const uint32_t* info_codes,
         RAISE_ADBC(
             AdbcConnectionGetInfoAppendString(array, info_codes[i], "PostgreSQL", error));
         break;
-      case ADBC_INFO_VENDOR_VERSION:
-        RAISE_ADBC(AdbcConnectionGetInfoAppendString(
-            array, info_codes[i], std::to_string(PQlibVersion()).c_str(), error));
+      case ADBC_INFO_VENDOR_VERSION: {
+        const char* stmt = "SHOW server_version_num";
+        auto result_helper = PqResultHelper{conn_, std::string(stmt), error};
+        RAISE_ADBC(result_helper.Prepare());
+        RAISE_ADBC(result_helper.Execute());
+        auto it = result_helper.begin();
+        if (it == result_helper.end()) {
+          SetError(error,
+                   "[libpq] PostgreSQL returned no rows for 'SELECT CURRENT_SCHEMA'");
```

Comment:
```suggestion
                   "[libpq] PostgreSQL returned no rows for '%s'", stmt);
```

Reply:
Ugh sorry about this - should have noticed already. Nice catch